### PR TITLE
feat: make polymarket notifier serverless

### PR DESCRIPTION
**Pull request by md0x**
**Motivation**

The previous design wasn't compatible with serverless bots configs as this bot was running in a continuous process.

**Summary**

- Filter ended markets from markets query
- Parallelise ancillary on-chain requests
- Simplify bot structure

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested
Fixes #XXXX

### CODE REVIEW

File: packages/monitor-v2/src/monitor-polymarket/index.ts [hunk 1]
```diff
@@ -1,50 +1,19 @@
...
+  logger.debug({
     at: "PolymarketMonitor",
     message: "Polymarket Monitor started 🔭",
-    botModes: params.botModes,
   });
 
-  const cmds = {
-    transactionsProposedEnabled: monitorTransactionsProposedOrderBook,
-  };
-
```
Comment (chrismaree) [inline packages/monitor-v2/src/monitor-polymarket/index.ts:20]: why was this removed? I think we want the ability to run the bot in _both_ looping mode _and_ serverless mode, no? for example we want to enable other community members or others to run the bot within a normal vm. this is how all other UMA/Across infra is structured.

File: packages/monitor-v2/src/monitor-polymarket/common.ts [hunk 2]
```diff
@@ -166,13 +155,23 @@ export const getPolymarketMarkets = async (params: MonitoringParams): Promise<Po
...
         endDate
+        umaResolutionStatus
       }
     }
   `;
 
   const { markets: polymarketContracts } = (await request(params.graphqlEndpoint, query)) as any;
 
-  return polymarketContracts.map((contract: { [k: string]: any }) => ({
+  // Remove markets with 1 week old endDate or more. So we only monitor markets that ended in the last week
```
Comment (chrismaree) [inline packages/monitor-v2/src/monitor-polymarket/common.ts:165]: this seems totally reasonable.

File: packages/monitor-v2/src/monitor-polymarket/index.ts (hunk 1 above)
Comment (md0x) [reply packages/monitor-v2/src/monitor-polymarket/index.ts:20]: I see. I removed it because this bot is not meant to be run in our infra like that (that's what initially confused me when I thought I could run them in loops)... But you are right. Let me undo this changes so we can support this way of running it.
